Cache/Requests.lean

```diff
@@ -128,20 +139,37 @@ def getRemoteRepo (mathlibDepPath : FilePath) : IO RepoInfo := do
     let isSpecialBranch := branchName == "master" || branchName.startsWith "bump/" ||
                           branchName.startsWith "nightly-testing"
 
-    -- Check if the current branch is tracking a PR
+    -- Check if the current commit coincides with any PR ref
     if !isSpecialBranch then
-      let prInfo ← IO.Process.output
-        {cmd := "gh", args := #["pr", "view", "--json", "headRefName,headRepositoryOwner,number"], cwd := mathlibDepPath}
-      unless prInfo.exitCode != 0 do
-        if let .ok json := Lean.Json.parse prInfo.stdout.trim then
-          if let .ok owner := json.getObjValAs? Lean.Json "headRepositoryOwner" then
-            if let .ok login := owner.getObjValAs? String "login" then
-              if let .ok repoName := json.getObjValAs? String "headRefName" then
-                if let .ok prNumber := json.getObjValAs? Nat "number" then
-                  let repo := s!"{login}/mathlib4"
-                  IO.println s!"Using cache from PR #{prNumber} source: {login}/{repoName}"
-                  let useFirst := if owner == "leanprover-community" then false else true
-                  return {repo := repo, useFirst := useFirst}
+      let mathlibRemoteName ← findMathlibRemote mathlibDepPath
+      let currentCommit ← IO.Process.output
+        {cmd := "git", args := #["rev-parse", "HEAD"], cwd := mathlibDepPath}
+
+      if currentCommit.exitCode == 0 then
+        let commit := currentCommit.stdout.trim
+        -- Get all PR refs that contain this commit
+        let prRefPattern := s!"refs/remotes/{mathlibRemoteName}/pr/*"
```

Comment:
This was broken in the scenario where you do `gh checkout pr XXXXX` (thought I should be doubled checked). The flag `--points-at` did return something. 

Since this is nested, we won't pull the right cache with the logic below under these circumstances. (Hopefully I am proven wrong...)